src/core_atmosphere/dynamics/mpas_atm_boundaries.F

```diff
@@ -225,15 +289,58 @@ subroutine mpas_atm_update_bdy_tend(clock, streamManager, block, firstCall, ierr
 
 
             dt = 1.0_RKIND / dt
-            lbc_tend_u(:,:) = (u(:,:) - lbc_tend_u(:,:)) * dt
-            lbc_tend_ru(:,:) = (ru(:,:) - lbc_tend_ru(:,:)) * dt
-            lbc_tend_rho_edge(:,:) = (rho_edge(:,:) - lbc_tend_rho_edge(:,:)) * dt
-            lbc_tend_w(:,:) = (w(:,:) - lbc_tend_w(:,:)) * dt
-            lbc_tend_theta(:,:) = (theta(:,:) - lbc_tend_theta(:,:)) * dt
-            lbc_tend_rtheta_m(:,:) = (rtheta_m(:,:) - lbc_tend_rtheta_m(:,:)) * dt
-            lbc_tend_rho_zz(:,:) = (rho_zz(:,:) - lbc_tend_rho_zz(:,:)) * dt
-            lbc_tend_rho(:,:) = (rho(:,:) - lbc_tend_rho(:,:)) * dt
-            lbc_tend_scalars(:,:,:) = (scalars(:,:,:) - lbc_tend_scalars(:,:,:)) * dt
+
+            !$acc parallel default(present)
+            !$acc loop gang vector collapse(2)
+            do iEdge=1,nEdges+1
+                do k=1,nVertLevels
+                    lbc_tend_u(k,iEdge) = (u(k,iEdge) - lbc_tend_u(k,iEdge)) * dt
+                    lbc_tend_ru(k,iEdge) = (ru(k,iEdge) - lbc_tend_ru(k,iEdge)) * dt
+                end do
+            end do
+
+            !$acc loop gang vector collapse(2)
+             do iEdge=1,nEdges+1
+                 do k=1,nVertLevels
+                     lbc_tend_rho_edge(k,iEdge) = (rho_edge(k,iEdge) - lbc_tend_rho_edge(k,iEdge)) * dt
+                 end do
+             end do
+
+            !$acc loop gang vector collapse(2)
+            do iCell=1,nCells+1
+                do k=1,nVertLevels+1
+                    lbc_tend_w(k,iCell) = (w(k,iCell) - lbc_tend_w(k,iCell)) * dt
+                end do
+            end do
+
+            !$acc loop gang vector collapse(2)
+            do iCell=1,nCells+1
+                do k=1,nVertLevels
+                    lbc_tend_theta(k,iCell) = (theta(k,iCell) - lbc_tend_theta(k,iCell)) * dt
+                    lbc_tend_rtheta_m(k,iCell) = (rtheta_m(k,iCell) - lbc_tend_rtheta_m(k,iCell)) * dt
+                end do
+            end do
+
+            !$acc loop gang vector collapse(2)
+            do iCell=1,nCells+1
+                do k=1,nVertLevels
+                    lbc_tend_rho_zz(k,iCell) = (rho_zz(k,iCell) - lbc_tend_rho_zz(k,iCell)) * dt
+                    lbc_tend_rho(k,iCell) = (rho(k,iCell) - lbc_tend_rho(k,iCell)) * dt
+                end do
+            end do
+            !$acc end parallel
+
+            !$acc parallel default(present)
```

Comment:
Could this `parallel` region be combined with the `parallel` region above it?

Reply:
Done.